### PR TITLE
fix: Char covert to String on remove specified unicode

### LINKED ISSUE #1638: Exception while inserting special char values like '\\u0000'
### Problem Type (问题类型)

Exception while inserting special chars

### Before submit

- [X] 我已经确认现有的 [Issues](https://github.com/hugegraph/hugegraph/issues) 与 [FAQ](https://hugegraph.github.io/hugegraph-doc/guides/faq.html) 中没有相同 / 重复问题

### Environment (环境信息)

- Server Version: v0.11.x
- Backend: RocksDB x nodes, HDD or SSD 
- OS: xx CPUs, xx G RAM, Centos 7.x 
- Data Size:  xx vertices, xx edges 

### Your Question (问题描述)

When i try to push the constant value '\\u0000' through JAVA API ,it results in java.lang.IllegalArgumentException.How to insert these type of values in hugegraph?

### Vertex/Edge example (问题点 / 边数据举例)

```javascript
[
  {
    "label": "CONSTANT",
    "id": 1,
    "properties": {
      "VALUE": "\u0000",
      "CLASS_NAME": "Ljava/lang/String;"
    }
  }
]
```

### Schema [VertexLabel, EdgeLabel, IndexLabel] (元数据结构)

_No response_

### PR DESCRIPTION
**Pull request by coderzc**
fixed #1638

### CODE REVIEW

**Review by javeme — APPROVED** [review]

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphIndexTransaction.java [hunk 1]
```diff
@@ -887,9 +896,9 @@ private boolean matchSearchIndexWords(String propValue, String fieldValue) {
 
     private Set<String> segmentWords(String text) {
         Set<String> words = this.textAnalyzer.segment(text);
-        for (char ch = INDEX_SYM_ENDING; ch <= INDEX_SYM_MAX; ch++) {
-            words.remove(ch);
-        }
+
+        // Ignore unicode \u0000 to \u0003
+        words.removeAll(IGNORE_SYM_SET);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphIndexTransaction.java:901]: can add a test case?

Comment (coderzc) [reply hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphIndexTransaction.java:901]: ok

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java [hunk 2]
```diff
@@ -8582,6 +8582,29 @@ public void testQueryByHasIdEmptyListInPage() {
         Assert.assertNull(page);
     }
 
+    @Test
+    public void testIgnoreSymbol() {
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java:8586]: rename to test testQueryBySearchIndexWithSpecialSymbol

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphIndexTransaction.java [hunk 3]
```diff
@@ -103,6 +103,15 @@
     public static final String INDEX_SYM_NULL = "\u0001";
     public static final String INDEX_SYM_EMPTY = "\u0002";
     public static final char INDEX_SYM_MAX = '\u0003';
+    public static final Set<String> IGNORE_SYM_SET;
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphIndexTransaction.java:106]: add `fixed #1638` to commit message

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java [hunk 4]
```diff
@@ -8582,6 +8582,29 @@ public void testQueryByHasIdEmptyListInPage() {
...
+        HugeGraph graph = graph();
+
+        graph.schema().indexLabel("personByName").onV("person")
+             .by("name").search().ifNotExist().create();
+
+        Vertex vertex = graph.addVertex(T.label, "person", "name",
+                                        "xyz\u0000abc", "city", "Hongkong",
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java:8593]: also test `\u0003`

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java [hunk 5]
```diff
@@ -8582,6 +8582,29 @@ public void testQueryByHasIdEmptyListInPage() {
...
+                                        "age", 15);
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java:8594]: what's the behavior if adding a vertex with `"name": "\u0000"`

**Review by javeme — DISMISSED** [review]

**Comment by javeme** [comment]
LGTM

**Review by imbajin — DISMISSED** [review]

**Comment by coderzc** [comment]
mysql&postgresql ci failed
```
Error:  testQueryBySearchIndexWithSpecialSymbol(com.baidu.hugegraph.core.VertexCoreTest)  Time elapsed: 0.304 s  <<< ERROR!
com.baidu.hugegraph.backend.BackendException: Failed to commit
	at com.baidu.hugegraph.core.VertexCoreTest.testQueryBySearchIndexWithSpecialSymbol(VertexCoreTest.java:8604)
Caused by: java.sql.BatchUpdateException: Batch entry 0 INSERT INTO g_ai (FIELD_VALUES, INDEX_LABEL_ID, ELEMENT_IDS, EXPIRED_TIME) VALUES ('xyz', 276, 'S1454:xyzS1454:xyzabc', 0) ON CONFLICT (FIELD_VALUES, INDEX_LABEL_ID, ELEMENT_IDS) DO UPDATE SET FIELD_VALUES = 'xyz', INDEX_LABEL_ID = 276, ELEMENT_IDS = 'S1454:xyzS1454:xyzabc', EXPIRED_TIME = 0 was aborted: ERROR: invalid byte sequence for encoding "UTF8": 0x00  Call getNextException to see other errors in the batch.
	at com.baidu.hugegraph.core.VertexCoreTest.testQueryBySearchIndexWithSpecialSymbol(VertexCoreTest.java:8604)
Caused by: org.postgresql.util.PSQLException: ERROR: invalid byte sequence for encoding "UTF8": 0x00
	at com.baidu.hugegraph.core.VertexCoreTest.testQueryBySearchIndexWithSpecialSymbol(VertexCoreTest.java:8604)
```

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java [hunk 6]
```diff
@@ -8590,10 +8590,10 @@ public void testQueryBySearchIndexWithSpecialSymbol() {
              .by("name").search().ifNotExist().create();
 
         Vertex vertex = graph.addVertex(T.label, "person", "name",
-                                        "xyz\u0000abc", "city", "Hongkong",
+                                        "xyz\u0002abc", "city", "Hongkong",
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java:8593]: also add case for "xyz\u0001abc"

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java [hunk 7]
```diff
@@ -8590,10 +8590,10 @@ public void testQueryBySearchIndexWithSpecialSymbol() {
...
                                         "age", 15);
         Vertex vertex2 = graph.addVertex(T.label, "person", "name",
-                                         "\u0000", "city", "Hongkong",
+                                         "\u0002", "city", "Hongkong",
```
Comment (javeme) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java:8596]: also add case for "\u0001"

**Comment by coderzc** [comment]
ci failed
```
Error:  Tests run: 714, Failures: 0, Errors: 1, Skipped: 13, Time elapsed: 265.664 s <<< FAILURE! - in com.baidu.hugegraph.core.CoreTestSuite
Error:  testQueryBySearchIndexWithSpecialSymbol(com.baidu.hugegraph.core.VertexCoreTest)  Time elapsed: 0.151 s  <<< ERROR!
java.lang.IllegalArgumentException: Illegal value of index property: ''
	at com.baidu.hugegraph.core.VertexCoreTest.testQueryBySearchIndexWithSpecialSymbol(VertexCoreTest.java:8610)
```

**Comment by javeme** [comment]
need to rebase master

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java [hunk 8]
```diff
@@ -61,6 +61,15 @@
     public static final String INDEX_VALUE_NULL = new String("<null>");
     public static final String INDEX_VALUE_EMPTY = new String("<empty>");
 
+    public static final Set<String> IGNORE_SYM_SET;
+    static {
+        List<String> list = new ArrayList<>(INDEX_SYM_MIN - INDEX_SYM_MAX);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java:66]: error `Illegal Capacity: -3`, expect `INDEX_SYM_MAX - INDEX_SYM_MIN`

Comment (coderzc) [reply hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java:66]: ok

**Review by javeme — APPROVED** [review]

**Review by imbajin — APPROVED** [review]

**Pull request merged**